src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadSegmentsToZipArchiveWriter.cs

```diff
@@ -41,18 +42,14 @@ public class IntegrationRoadSegmentsToZipArchiveWriter : IZipArchiveWriter<Edito
 
         const int integrationBufferInMeters = 350;
 
-        var segmentsInContour = await context.RoadSegments
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var segmentsInContour = await context.RoadSegments.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
         var geometryForSegmentsInContour = GeometryConfiguration.GeometryFactory
             .BuildGeometry(segmentsInContour.Select(segment => segment.Geometry));
 
         var boundaryForSegments = geometryForSegmentsInContour.ConvexHull();
         var boundaryWithIntegrationBuffer = boundaryForSegments.Buffer(integrationBufferInMeters);
 
-        var segmentsInIntegrationBuffer = await context.RoadSegments
-            .InsideContour(boundaryWithIntegrationBuffer as IPolygonal)
-            .ToListAsync(cancellationToken);
+        var segmentsInIntegrationBuffer = await context.RoadSegments.ToListWithPolygonials(boundaryWithIntegrationBuffer as IPolygonal, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```

Comment:
use a line-separated method chain (like what it was)

Reply:
Done